src/index.ts

```diff
@@ -336,8 +344,27 @@ const handleToDonations = async ({
     item => item.txHash === transactionHash && item.amountRemaining === '0' && item.isReturn === isReturn,
   );
 
-  const toDonation = toIndex !== -1 ? toNotFilledDonationList.splice(toIndex, 1)[0] : undefined;
-  if (toDonation === undefined) {
+  let toDonation = toIndex !== -1 ? toNotFilledDonationList.splice(toIndex, 1)[0] : undefined;
+  if (!toDonation && !isReturn ){
+    // If we didnt find any donation in above statement we should check to see if we find
+    // donations with isReturn===true, so in this case it's ok to donation have different isReturn
+    // if it's a donation with the trace donation parent, and it's ownerType is campaign
+    const toIndex= toNotFilledDonationList.findIndex(
+      item => item.txHash === transactionHash &&
+        item.amountRemaining === '0' &&
+        item.isReturn === true,
+    );
+    toDonation = toIndex !== -1 ? toNotFilledDonationList.splice(toIndex, 1)[0] : undefined;
+    const isDonationBackToCampaign = toDonation && await isDonationBackToCampaignFromTrace(toDonation)
```

Comment:
We computed this logic (lines 358:364) once here and once in lines 567:571

Reply:
There is two different logic, in L358 we check when no toDonaiton found, but in L567, we check when toDonation found, so it's not duplicated

Reply:
@mohammadranjbarz you are right, thanks.